packages/web/src/fauceting/PhoneInput.tsx

```diff
@@ -0,0 +1,330 @@
+
+    if (phoneInfo) {
+      this.setState({ phoneNumber: phoneInfo.displayNumber })
+      this.props.onChangeNumber(phoneInfo.e164Number)
+    } else {
+      this.setState({ phoneNumber: rawPhone })
+      this.props.onChangeNumber('')
+    }
+  }
+  updateSuggestions = ({ value }: { value: string; reason: UpdateReasons }) => {
+    this.onChangeCountryQuery(value)
+  }
+
+  onInputChange = (_, { newValue }: { newValue: string }) => {
+    this.onChangeCountryQuery(newValue)
+
+    // use Set Immediate to avoid typing lag
+    setImmediate(() => {
+      const country = COUNTRIES.getCountry(newValue)
+      if (country.displayName) {
+        // @ts-ignore -- TS isnt recognizing that Localized country extends Country\
+        const countryCode = country.alpha2
+        this.setState({
+          countryQuery: country.displayName,
+          countryCode,
+        })
+      } else {
+        this.setState({
+          countryCode: '',
+          phoneNumber: '',
+        })
+        this.props.onChangeNumber('')
+      }
+    })
+  }
+  renderSuggestionsContainer = ({ containerProps, children }: SuggestionContainerProps) => {
+    const { className, id, ...otherProps } = containerProps
+    const isOpen = this.filteredCountries().length > 0
+
+    return (
+      <View
+        {...otherProps}
+        nativeID={id}
+        style={[styles.suggestions, isOpen && styles.suggestionsOpen]}
+      >
+        {children}
+      </View>
+    )
+  }
+
+  renderTextInput = (props: AutoSuggestInputProps) => {
+    return (
+      <View style={styles.countryInputContainer}>
+        <View style={styles.emojiContainer}>
+          <Text>{this.getEmoji()}</Text>
+        </View>
+        <TextInput
+          style={[
+            standardStyles.input,
+            standardStyles.inputDarkMode,
+            styles.countryInput,
+            this.getEmoji().length && styles.countryInputWithEmoji,
+          ]}
+          focusStyle={standardStyles.inputDarkFocused}
+          {...props}
+          value={this.state.countryQuery}
+        />
+      </View>
+    )
+  }
+
+  renderItem = (countryCode: string) => {
+    // @ts-ignore -- TS isnt recognizing that Localized country extends Country
+    const { displayName, emoji } = COUNTRIES.getCountryByCode(countryCode)
+
+    const onPress = () => {
+      this.setState({ countryCode, phoneNumber: '' })
+      this.onChangeCountryQuery(displayName)
+    }
+
+    return (
+      <TouchableOpacity onPress={onPress}>
+        <View style={[standardStyles.row, styles.item]}>
+          <Text>{emoji || `🏳`}</Text>
+          <Text style={[fonts.legal, textStyles.invert]}>{displayName}</Text>
+        </View>
+      </TouchableOpacity>
+    )
+  }
+
+  render() {
+    const inputProps: InputProps = {
+      placeholder: this.props.t('countryPlaceholder'),
+      value: this.state.countryQuery,
+      onChange: this.onInputChange,
+      type: 'search',
+    }
+
+    return (
+      <>
+        <style>
+          {`
```

Comment:
Maybe I'm missing some context on the web proj but is there a reason to have these styles inlined instead of included at the bottom in TS?

Reply:
For this 3rd party component there are some elements it creates that i dont have direct access to only to these css classes.